app/src/pages/IndividualRequest.tsx

```diff
@@ -37,66 +37,72 @@ const IndividualRequest = () => {
-          const approvalFileArray : Array<IFile> = [];
-
-          if (reimbursementRequest.purchases.length > 0){
-            reimbursementRequest.purchases.forEach((purchase, index) => {
-              if (purchase.fileObj){
-                purchaseFileArray.splice(index, 0, purchase.fileObj);
-              }
-            });
-          }
-
-          if (reimbursementRequest.approvals){
-            reimbursementRequest.approvals.forEach((approval, index) => {
-              if (approval.fileObj){
-                approvalFileArray.splice(index, 0, approval.fileObj);
-              }
-            });
-          }
-
-          // Set new states
-          setReimbursementRequest(reimbursementRequest);
-          setRequestState(reimbursementRequest.state);
-          setPurchases(reimbursementRequest.purchases);
-          setPurchaseFiles(purchaseFileArray);
-          
-          if (reimbursementRequest.approvals){
-            setApprovals(reimbursementRequest.approvals);
-            setApprovalFiles(approvalFileArray)
-          }
+      }
+      let response = await axios(axiosReqConfig);
+      if (response.status === 200) {
+        
+        // Populate values with existing record
+        const reimbursementRequest: ReimbursementRequest = response.data;
+        const purchaseFileArray : Array<IFile> = [];
+        const approvalFileArray : Array<IFile> = [];
+
+        if (reimbursementRequest.purchases.length > 0){
+          reimbursementRequest.purchases.forEach((purchase, index) => {
+            if (purchase.fileObj){
+              purchaseFileArray.splice(index, 0, purchase.fileObj);
+            }
+          });
+        }
+
+        if (reimbursementRequest.approvals){
+          reimbursementRequest.approvals.forEach((approval, index) => {
+            if (approval.fileObj){
+              approvalFileArray.splice(index, 0, approval.fileObj);
+            }
+          });
+        }
+
+        // Set new states
+        setReimbursementRequest(reimbursementRequest);
+        setRequestState(reimbursementRequest.state);
+        setPurchases(reimbursementRequest.purchases);
+        setPurchaseFiles(purchaseFileArray);
+        
+        if (reimbursementRequest.approvals){
+          setApprovals(reimbursementRequest.approvals);
+          setApprovalFiles(approvalFileArray)
+        }
+
+        // Determine locked status of fields. If not admin and the state of request is not INCOMPLETE, lock the fields.
+        if (!isAdmin && reimbursementRequest.state !== RequestStates.INCOMPLETE){
+          setLocked(true);
         }
-      } catch (e) {
-        console.warn('Record could not be retrieved.');
       }
-    })();
+    } catch (e) {
+      console.warn('Record could not be retrieved.');
+    }
   }, []);
 
   // Fired when the record is updated (i.e. User selects UPDATE.)
   const handleUpdate = async () => {
     // TODO: If approvals or purchases don't change, don't send back that info
```

Comment:
Memoizing here could satisfy the TODO

Reply:
There's a bigger change to how I want to load and save the files in the backlog. I think I'll try to work this in with that.